### PR TITLE
Searching files functionality. 

### PR DESCRIPTION
**Pull request by Keworker**
Add searching files by name (sync). Invalid codec right now.

We steel need to do something:

- Fix codecs. 
- Add searching in file content. 
- Make searching async. 
- Fix UI.

### CODE REVIEW

**Comment by Keworker** [comment]
It would be good to convert 3 flags to one enum. Also it would be better to use `switch/case` in this way.

**Comment by Keworker** [comment]
Learn about queues in Python's multiprocessing. 
https://www.geeksforgeeks.org/python-multiprocessing-queue-vs-multiprocessing-manager-queue/

**Comment by Keworker** [comment]
Work with files and don't depend on encoding. 
https://stackoverflow.com/questions/34428007/read-file-with-python-without-knowing-encoding

**Comment by Keworker** [comment]
Error: if there is an empty element in user's sequence program will crash.

**Comment by Keworker** [comment]
Error: `.pyc` outputted with searching `.py`

**Comment by Keworker** [comment]
Fix sorting.

**Comment by Keworker** [comment]
@Qarru , please, re-intstall your MIUI and review this PR. It's waits only for async work, fix of results sorting and **your review**!

**Comment by Keworker** [comment]
It's important to profile searching: maybe async work won't make it faster (memory load is important, not CPU).

**Comment by Keworker** [comment]
May close "Pull request template #14"

**Comment by Keworker** [comment]
This PR resolves "About page #10" now.

File: src/python/data/File.py [hunk 1]
```diff
@@ -6,4 +6,8 @@ class File:  # {
     path: str
     matchCount: int
     preview: str
+
+    def __lt__(self, other):  # {
```
Comment (Qarru) [inline src/python/data/File.py:10]: _ _cmp_ _ ?

File: src/python/data/FilenameMode.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+
+class FilenameMode(Enum):  # {
+    EXTENSION: bool = 0
```
Comment (Qarru) [inline src/python/data/FilenameMode.py:5]: var: bool = 0...

Comment (Qarru) [reply src/python/data/FilenameMode.py:5]: in case of future filenamemode extension, you probably should use int

**Review by Qarru — CHANGES_REQUESTED** [review]

File: README.md
[the remark this answers is not in the document]
Comment (Keworker) [reply README.md:1]: I choose image that i like. Now i add picture to text, maybe it looks better. I won't change font.

File: src/python/data/FilenameMode.py (hunk 2 above)
Comment (Keworker) [reply src/python/data/FilenameMode.py]: You are right. I think typing for enums is useless at all, so, i deleted it.

File: src/python/AboutWindow.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+from PyQt6.QtCore import QRect, Qt
+from PyQt6.QtGui import QGuiApplication, QIcon, QPixmap, QFont
+from PyQt6.QtWidgets import QWidget, QHBoxLayout, QLabel, QSizePolicy, QVBoxLayout
+
+from src.res.strings import ABOUT_TITLE, APP_TITLE, VERSION, ABOUT_TEXT, REPO_LINK, CONTACTS
+
+
+class AboutWindow(QWidget):  # {
+    def __init__(self, application: QGuiApplication, iconSmall: QIcon, iconLarge: QPixmap):  # {
```
[the remark this answers is not in the document]
Comment (Keworker) [reply src/python/AboutWindow.py:9]: Ok, fixed.

File: src/python/data/File.py (hunk 1 above)
Comment (Keworker) [reply src/python/data/File.py:10]: I think you are old school python developer :D 
`__cmp__` is no longer supported in Python 3, `__lt__` is normal solution now.

File: .github/PULL_REQUEST_TEMPLATE.md
[the remark this answers is not in the document]
Comment (Keworker) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Is that variant better ? 
```md
# About Pull Request

\# Write about your PR here

# Successful PR check-list

- [ ] I explained what's changed. 
- [ ] The PR is ready to merge and completely finished (or I've written out what I'm going to do next in this PR and moved it to draft). 
- [ ] I've requested a review from the project owner or some of the contributors. 
- [ ] I've read [contributing guide](https://github.com/Keworker/file-finder-pyqt/blob/master/CONTRIBUTING.md). 
- [ ] I understand that the PR will not get a review until all checks have been passed. 
- [ ] I understand that I can't merge PR without a review. 
```

File: src/python/data/File.py (hunk 1 above)
Comment (Qarru) [reply src/python/data/File.py:10]: o-ops. Fine then

File: README.md
[the remark this answers is not in the document]
Comment (Qarru) [reply README.md:1]: fine

**Review by Qarru — APPROVED** [review]
:thumb_up

File: .github/PULL_REQUEST_TEMPLATE.md
[the remark this answers is not in the document]
Comment (Keworker) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: @Qarru

**Review by Qarru — APPROVED** [review]

**Pull request merged**